app/scripts/components/exploration/components/timeline/timeline-head.tsx

```diff
@@ -32,25 +20,136 @@ interface TimelineHeadBaseProps {
   isStrokeDashed?: boolean;
   children: React.ReactNode;
   'data-tour'?: string;
+  xPosOffset?: number;
+  zIndex?: number;
 }
 
+const TimelineHeadWrapper = styled.div`
+  position: absolute;
+  right: ${RIGHT_AXIS_SPACE - SVG_PADDING}px;
+  top: -1rem;
+  height: calc(100% + 1rem);
+  pointer-events: none;
+  z-index: ${themeVal('zIndices.overlay' as any)};
+`;
+
+const TimelinePlayheadWrapper = styled.div`
+  position: absolute;
+
+  &:hover {
+    cursor: grab;
+  }
+
+  &.playhead-grab * {
+    cursor: grabbing;
+    background-color: ${TIMELINE_PLAYHEAD_COLOR_PRIMARY} !important;
+
+    &::after {
+      border-top: 8px solid ${TIMELINE_PLAYHEAD_COLOR_PRIMARY};
+    }
+  }
+`;
+
+const TimelinePlayheadBase = styled.div`
+  background-color: ${TIMELINE_PLAYHEAD_COLOR_SECONDARY};
+  color: ${TIMELINE_PLAYHEAD_COLOR_TEXT};
+  padding: 3px;
+  border-radius: 4px;
+  font-size: 0.75rem;
+  position: relative;
+`;
+
+const TimelinePlayheadLeft = styled(TimelinePlayheadBase)`
+  background-color: ${TIMELINE_PLAYHEAD_COLOR_PRIMARY};
+  border-top-left-radius: 4px;
+  border-top-right-radius: 4px;
+  border-bottom-left-radius: 4px;
+  border-bottom-right-radius: 0;
+  position: absolute;
+  transform: translateX(-100%);
+  white-space: nowrap;
+  left: 17px;
```

Comment:
I am a bit concerned about this magic number. Why should this be 17px?

Reply:
I went through the styles and changed up the ones I could. Some magic numbers are there as we're implementing the playheads in the context of an svg canvas, part of which is not as responsive.

Some styles (e.g. the pseudoselector styling for :after, :before for the indicator triangles) I kept in 'px' units as I think it's easier to control them from a dev perspective, but let me know wdyt.